src/OpenTelemetry.Extensions.Hosting/Implementation/TelemetryHostedService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// <copyright file="TelemetryHostedService.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Extensions.Hosting.Implementation
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using OpenTelemetry.Trace.Configuration;
+
+    internal class TelemetryHostedService : IHostedService
+    {
+        private readonly IServiceProvider serviceProvider;
+
+        public TelemetryHostedService(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            // Ensure the factory was created when the app starts.
+            // This will create and start any configured instrumentations.
+            this.serviceProvider.GetService<OpenTelemetrySdk>();
```

Comment:
@cijothomas Probably better to use GetRequiredService here. Mine used GetService because I dropped it into the old HostedService and it needed to live side-by-side with the TracerFactory. Since you split it into its own service, GetRequiredService will add in a little protection by throwing if someone managed to mess things up by not registering an OpenTelemetrySdk.